frontend/src/components/regions/UserActions/index.jsx

```diff
@@ -22,17 +23,54 @@ const days = [
   "Saturday",
 ];
 
+const matchEventTypesWithColors = ({ eventTypes, eventTypeColors }) => {
+  const arr = [];
+  eventTypes = Object.keys(eventTypes).map((key) => eventTypes[key]);
+
+  eventTypes.forEach((eventType) => {
+    arr.push({
+      id: eventType.id,
+      eventName: eventType.name,
+      eventColor: eventTypeColors[eventType.name],
+    });
+  });
+  return arr;
+};
+
+function mapData({ eventTypesWithColors, activityLogEntries }) {
+  activityLogEntries = Object.keys(activityLogEntries).map(
+    (key) => activityLogEntries[key]
+  );
+
+  const newData = activityLogEntries.map((item, index) => {
+    const data = eventTypesWithColors.find(
+      (elem) => elem.id === item.eventType
+    );
+    return {
+      ...item,
+      eventName: data ? data.eventName : "",
+      eventColor: data ? data.eventColor : "",
+    };
+  });
+  return newData;
+}
+
 function UserActionsUnconnected({
   authedUserId,
   projectReviews,
   cardSummaries,
+  activityLogEntries,
   fetchProjectReviewsPages,
   fetchCardCompletions,
   userBurndownStats,
   fetchUserBurndownStats,
+  fetchActivityLogEntries,
+  eventTypes,
```

Comment:
small placement thing here. place the mapStateToProps variables at the top to keep consistency